### ISSUE TITLE
Can we give compile-time error for panic cases for `module constant declaration`?

### ISSUE THREAD

**Issue by prakanth97** [ballerina-platform/ballerina-spec#1168]
**Description:**
https://ballerina.io/spec/lang/master/#section_8.8 
```
A module-const-decl declares a compile-time constant.
```
https://ballerina.io/spec/lang/master/#section_6.23
https://ballerina.io/spec/lang/master/#section_6.22
According to the specs, `multiplicative expression` and `unary expression` will result runtime error for `overflow`, `division by zero` and etc. While `module-const-decl` declares a compile-time constant, do we need to give compile-time errors for panic cases?

   
**Code sample that shows issue:**
```
const NUM1 = -1;
const int NUM2 = -9223372036854775807 - 1;
const int ANS1 = NUM2 / NUM1;          // error
```

**Comment by hasithaa** [comment]
Yes, it should be a compile-time error.

**Comment by jclark** [comment]
I agree with @hasithaa. The value of a constant expression has to be known at compile-time, so it has to be a compile-time error. Furthermore, the type is a singleton type, and there is no possible singleton type in this case.
